### DISCUSSION TITLE
Solution of coupled time dependent nonlinear equations

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#17948] [Q&A Modules: General] [answered]
Is there an example MOOSE file that solves a problem similar to the following? :

d/dt (n1) = C1 x phi x n1
d/dt (n2) = C2 x phi x n1-C3 x phi x n2

where phi = phi(x,t)

**Comment by GiudGiud** [comment, marked as answer]
Hello

Do you want to solve for phi (variable) or have it fixed (function) ?
Not sure we have an example for this case exactly, but a combination of CoupledForce, TimeDerivative kernels (FE or FV) will do the trick.

best,
Guillaume

**Reply by TLWise** [reply]
I would in a simple case want phi to be a constant value.

Thank you.

**Reply by GiudGiud** [reply]
The kernels I mentioned can be used for your case then. 
You can adapt [CoefReaction](https://mooseframework.inl.gov/source/kernels/CoefReaction.html) to make it use a function or store phi as a material property (using a GenericFunctionMaterial) and use [MatReaction](https://mooseframework.inl.gov/source/kernels/MatReaction.html).
